src/System.Windows.Forms/src/System/Windows/Forms/WindowsFormsSynchronizationContext.cs

```diff
@@ -43,20 +43,12 @@ private WindowsFormsSynchronizationContext(Control? marshalingControl, Thread? d
     // Directly holding onto the Thread can prevent ThreadStatics from finalizing.
     private Thread? DestinationThread
     {
-        get
-        {
-            if ((_destinationThread is not null) && _destinationThread.IsAlive)
-            {
-                return _destinationThread.Target as Thread;
-            }
-
-            return null;
-        }
+        get => _destinationThread?.TryGetTarget(out Thread? target) == true ? target : null;
         set
         {
             if (value is not null)
             {
-                _destinationThread = new WeakReference(value);
+                _destinationThread = new(value);
```

Comment:
Perhaps we want to reuse an existing weakref object? Something like

```suggestion
                if (_destinationThread is null)
                {
                    _destinationThread = new(value);
                }
                else
                {
                    _destinationThread.SetTarget(value);
                }
```